src/client/content_mapblock.cpp

```diff
@@ -224,7 +266,7 @@ void MapblockMeshGenerator::drawCuboid(const aabb3f &box,
 {
 	assert(tilecount >= 1 && tilecount <= 6); // pre-condition
 
-	auto vertices = setupCuboidVertices(box, txc, tiles, tilecount);
+	auto vertices = setupCuboidVertices(box, txc, tiles, tilecount, cur_node.p);
```

Comment:
Btw. you could easily fix scale being limited to int fractions of 16 (currently it's mesh chunk size dependent then), by adding `blockpos_nodes` here. (Maybe follow up PR.)